file-sink/src/test/java/org/springframework/cloud/stream/module/file/FileSinkTests.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.file;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.io.FileReader;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.boot.test.WebIntegrationTest;
+import org.springframework.cloud.stream.messaging.Sink;
+import org.springframework.messaging.support.MessageBuilder;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.util.FileCopyUtils;
+
+/**
+ * @author Mark Fisher
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = FileSinkApplication.class)
+@WebIntegrationTest(randomPort = true)
+@DirtiesContext
+public abstract class FileSinkTests {
+
+	@Autowired
+	protected Sink sink;
+
+	@WebIntegrationTest({"name = test", "dir = /tmp/dataflow-tests", "suffix=txt"})
+	public static class TextTests extends FileSinkTests {
+
+		@Test
+		public void test() throws Exception {
+			sink.input().send(MessageBuilder.withPayload("this is a test").build());
+			File file = new File("/tmp/dataflow-tests/test.txt");
+			file.deleteOnExit();
```

Comment:
I need to have an explicitly named dir in the annotations. Would that be possible if a `TemporaryFolder` is the root of the path?